app/scripts/react/components/SmartSelect/index.jsx

```diff
@@ -84,19 +85,38 @@ export default class SmartSelect extends Component {
         this.resetRequestState(timeoutFailure);
       });
   }
+
+  onOpen() {
+    const dumb = this.refs.dumb;
+    const menu = dumb.refs.menu.getDOMNode();
+    if (!(menu instanceof HTMLElement)) {
+      return;
+    } 
+    
+    const menuRect = menu.getBoundingClientRect();
+    this.setState({ dropup: (menuRect.bottom > window.innerHeight) });
+  }
+
+  onClose() {
+    this.setState({ dropup: false });
+  }
   
   render() {
-    const { fieldName, options, dropup } = this.props;
+    const { fieldName, options } = this.props;
+    const { dropup, requestStatus: status, value } = this.state;
```

Comment:
const { dropup, value, requestStatus: status } = this.state;
https://github.com/airbnb/javascript#3.7